src/patternfly/components/CodeSnippet/code-snippet.scss

```diff
@@ -0,0 +1,132 @@
+.pf-c-code-snippet {
+  // controls
+  --pf-c-code-snippet__controls--Top: var(--pf-c-code-snippet__main--BorderWidth);
+  --pf-c-code-snippet__controls--c-button--m-control--Color: var(--pf-global--Color--200);
+  --pf-c-code-snippet__controls--c-button--m-control--hover--Color: var(--pf-global--Color--100);
+  --pf-c-code-snippet__controls--c-button--m-control--focus--Color: var(--pf-global--Color--100);
+  --pf-c-code-snippet__controls--c-button--m-control--disabled--after--BorderBottomColor: var(--pf-global--BorderColor--100);
+
+  // main
+  --pf-c-code-snippet__main--BorderColor: var(--pf-global--BorderColor--100);
+  --pf-c-code-snippet__main--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__main--BackgroundColor: var(--pf-global--BackgroundColor--100);
+  --pf-c-code-snippet--m-read-only__main--BackgroundColor: var(--pf-global--disabled-color--300);
+
+  // code
+  --pf-c-code-snippet__code--m-drag-hover--before--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__code--m-drag-hover--before--BorderColor: var(--pf-global--primary-color--100);
+  --pf-c-code-snippet__code--m-drag-hover--before--ZIndex: var(--pf-global--ZIndex--xs);
+  --pf-c-code-snippet__code--m-drag-hover--after--BackgroundColor: var(--pf-global--primary-color--100);
+  --pf-c-code-snippet__code--m-drag-hover--after--Opacity: .1;
+
+  // tab
+  --pf-c-code-snippet__tab--BackgroundColor: var(--pf-global--BackgroundColor--100);
+  --pf-c-code-snippet__tab--Color: var(--pf-global--Color--200);
+  --pf-c-code-snippet__tab--Top: var(--pf-c-code-snippet__main--BorderWidth);
+  --pf-c-code-snippet__tab--PaddingTop: var(--pf-global--spacer--form-element);
+  --pf-c-code-snippet__tab--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-code-snippet__tab--PaddingBottom: var(--pf-global--spacer--form-element);
+  --pf-c-code-snippet__tab--PaddingLeft: var(--pf-global--spacer--sm);
+  --pf-c-code-snippet__tab--BorderTopWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__tab--BorderRightWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__tab--BorderBottomWidth: 0;
+  --pf-c-code-snippet__tab--BorderLeftWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__tab--BorderColor: var(--pf-global--BorderColor--100);
+
+  // tab text
+  --pf-c-code-snippet__tab-icon--text--MarginLeft: var(--pf-global--spacer--sm);
+
+  width: 100%;
+
+  &.pf-m-read-only {
+    --pf-c-code-snippet__main--BackgroundColor: var(--pf-c-code-snippet--m-read-only__main--BackgroundColor);
+  }
+}
+
+.pf-c-code-snippet__header {
+  position: relative;
+  display: flex;
+  align-items: flex-end;
+}
+
+.pf-c-code-snippet__controls {
+  position: relative;
+  top: var(--pf-c-code-snippet__controls--Top);
```

Comment:
curious why you'd use `top` instead of a negative margin? This leaves space between the top of the element and whatever is before it, similar to using `translate()`. 

<img width="253" alt="Screen Shot 2020-09-09 at 5 02 14 PM" src="https://user-images.githubusercontent.com/35148959/92659606-514e1080-f2be-11ea-8e5a-bff08e79df6b.png">